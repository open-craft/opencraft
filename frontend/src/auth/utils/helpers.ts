import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
