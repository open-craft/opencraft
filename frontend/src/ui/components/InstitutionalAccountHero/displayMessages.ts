const messages = {
  institutionalAccountTitle: {
    defaultMessage: 'Institutional account',
    description: 'The title of the hero that promotes institutional instances.'
  },
  institutionalAccountText: {
    defaultMessage:
      'An institutional account gives you access to analytics, paid courses and ' +
      'certificates, mobile apps, priority support and more.',
    description:
      'Text that goes on the hero to promote institutional instances.'
  },
  contactUsButton: {
    defaultMessage: 'Contact Us',
    description: 'Text that goes in the contact us button.'
  },
  findOutMoreButton: {
    defaultMessage: 'Find out more',
    description: 'Text that goes in the find out more button.'
  }
};

export default messages;
