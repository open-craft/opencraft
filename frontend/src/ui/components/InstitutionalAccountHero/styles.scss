@import '~styles/theme';

.hero-container {
  background-image: url('~assets/institutional-bg.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  display: flex;
  margin: 0 -15px;
}

.hero-div {
    background-color: rgba(255, 255, 255, 0.8);
    margin: 50px auto;
    padding: 30px 20px;
    max-width: 80%;
    text-align: center;

    h2 {
      font-family: $playfair-font;
      font-size: 30px;
      font-weight: bold;
      letter-spacing: 1.8px;
      color: $primary-1;
      margin-bottom: 16px;
    }

    p {
      font-family: $chivo-font;
      font-size: 18px;
      color: #1e4c59;
    }

    @media (min-width: 800px)
    {
      max-width: 640px;
    }
}

.institutional-button-container {
  button {
    margin: 8px;
  }
}
