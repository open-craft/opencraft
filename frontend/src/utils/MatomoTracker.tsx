import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
