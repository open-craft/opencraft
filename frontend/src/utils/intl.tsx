import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
