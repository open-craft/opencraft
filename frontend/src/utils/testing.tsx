import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
