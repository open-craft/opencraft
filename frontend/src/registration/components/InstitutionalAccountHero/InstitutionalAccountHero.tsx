import * as React from 'react';
import { Button } from "react-bootstrap";
import { WrappedMessage } from "utils/intl";

import messages from './displayMessages';

import './styles.scss';

export const InstitutionalAccountHero: React.FC = () => (
    <div className="hero-container">
        <div className="hero-div">
            <h2>
              <WrappedMessage messages={messages} id="institutionalAccountTitle"/>
            </h2>
            <p>
              <WrappedMessage messages={messages} id="institutionalAccountText"/>
            </p>
            <div className="institutional-button-container">
              <Button variant="outline-primary" size="lg">
                <WrappedMessage messages={messages} id="contactUsButton"/>
              </Button>
              <Button variant="secondary" size="lg">
                <WrappedMessage messages={messages} id="findOutMoreButton"/>
              </Button>
            </div>
        </div>
    </div>
);
