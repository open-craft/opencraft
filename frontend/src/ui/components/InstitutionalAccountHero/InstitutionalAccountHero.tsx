import * as React from 'react';
import { CONTACT_US_LINK, ENTERPRISE_COMPARISON_LINK } from 'global/constants';
import { Button } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';

import messages from './displayMessages';

import './styles.scss';

export const InstitutionalAccountHero: React.FC = () => (
  <div className="hero-container">
    <div className="hero-div">
      <h2>
        <WrappedMessage messages={messages} id="institutionalAccountTitle" />
      </h2>
      <p>
        <WrappedMessage messages={messages} id="institutionalAccountText" />
      </p>
      <div className="institutional-button-container">
        <Button
          variant="outline-primary"
          size="lg"
          onClick={() => {
            window.location.href = CONTACT_US_LINK;
          }}
        >
          <WrappedMessage messages={messages} id="contactUsButton" />
        </Button>
        <Button
          variant="secondary"
          size="lg"
          onClick={() => {
            window.location.href = ENTERPRISE_COMPARISON_LINK;
          }}
        >
          <WrappedMessage messages={messages} id="findOutMoreButton" />
        </Button>
      </div>
    </div>
  </div>
);
