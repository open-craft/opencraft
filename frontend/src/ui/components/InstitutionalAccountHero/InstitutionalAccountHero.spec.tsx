import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { InstitutionalAccountHero } from './InstitutionalAccountHero';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<InstitutionalAccountHero />).toJSON();
    expect(tree).toMatchSnapshot();
});
