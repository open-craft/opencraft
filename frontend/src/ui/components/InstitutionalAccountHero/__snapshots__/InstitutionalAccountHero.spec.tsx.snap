// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="hero-container"
>
  <div
    className="hero-div"
  >
    <h2>
      Institutional account
    </h2>
    <p>
      An institutional account gives you access to analytics, paid courses and certificates, mobile apps, priority support and more.
    </p>
    <div
      className="institutional-button-container"
    >
      <button
        className="btn btn-outline-primary btn-lg"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Contact Us
      </button>
      <button
        className="btn btn-secondary btn-lg"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Find out more
      </button>
    </div>
  </div>
</div>
`;
